packages/protocol/src/action-protocol/edge-modification.ts

```diff
@@ -14,41 +14,98 @@
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ReconnectEdgeOperations`.
  */
-export class ReconnectEdgeOperation implements Operation {
-    static readonly KIND = 'reconnectEdge';
-    constructor(
-        public readonly edgeElementId: string,
-        public readonly sourceElementId: string,
-        public readonly targetElementId: string,
-        public readonly kind: string = ReconnectEdgeOperation.KIND
-    ) {}
+export interface ReconnectEdgeOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof ReconnectEdgeOperation.KIND;
+
+    /**
+     * The edge element that should be reconnected.
+     */
+    edgeElementId: string;
+
+    /**
+     * The (new) source element of the edge.
+     */
+    sourceElementId: string;
+
+    /**
+     * The (new) target element of the edge.
+     */
+    targetElementId: string;
+
+    /*
+     * Additional arguments for custom behavior.
+     */
+    args?: Args;
 }
 
-export function isReconnectEdgeOperation(action: any): action is ReconnectEdgeOperation {
-    return (
-        isActionKind(action, ReconnectEdgeOperation.KIND) &&
-        isString(action, 'edgeElementId') &&
-        isString(action, 'sourceElementId') &&
-        isString(action, 'targetElementId')
-    );
+export namespace ReconnectEdgeOperation {
+    export const KIND = 'reconnectEdge';
+
+    export function is(object: any): object is ReconnectEdgeOperation {
+        return (
+            Operation.hasKind(object, KIND) &&
+            hasStringProp(object, 'edgeElementId') &&
+            hasStringProp(object, 'sourceElementId') &&
+            hasStringProp(object, 'targetElementId')
+        );
+    }
+
+    export function create(options: {
+        edgeElementId: string;
+        sourceElementId: string;
+        targetElementId: string;
+        args?: Args;
+    }): ReconnectEdgeOperation {
+        return {
+            kind: KIND,
+            isOperation: true,
+            ...options
+        };
+    }
 }
 
 /**
  * An edge may have zero or more routing points that "re-direct" the edge between the source and the target element. In order to set these
  * routing points the client may send a `ChangeRoutingPointsOperation`.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ChangeRoutingPointsOperations`.
  */
-export class ChangeRoutingPointsOperation implements Operation {
-    static readonly KIND = 'changeRoutingPoints';
-    constructor(public newRoutingPoints: ElementAndRoutingPoints[], public readonly kind: string = ChangeRoutingPointsOperation.KIND) {}
+export interface ChangeRoutingPointsOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.